kfs-core/src/main/java/edu/arizona/kfs/sys/batch/service/impl/BuildingImportServiceImpl.java

```diff
@@ -309,76 +290,76 @@ private void updateBuilding(Building building, String routeCodeValue, String cam
 			reportMsg += "BuildingName(" + building.getBuildingName() + "), ";
 			building.setBuildingName(archBuilding.getBuildingName());
 		}
-		
+
 		if (StringUtils.isNotBlank(buildingExt.getRouteCode())) {
 			kfsBuild = buildingExt.getRouteCode();
 		}
-		else {
-			kfsBuild = "";
-		}
-		if (KFSParameterKeyConstants.HYPHEN.equalsIgnoreCase(routeCodeValue)) {
-			archBuild = "";
-		}
-		else {
+		if (!KFSParameterKeyConstants.HYPHEN.equalsIgnoreCase(routeCodeValue)) {
 			archBuild = routeCodeValue;
 		}
+
 		if (!archBuild.equalsIgnoreCase(kfsBuild)) {
 			if (!KFSParameterKeyConstants.HYPHEN.equalsIgnoreCase(routeCodeValue)) {
-				reportMsg += "RouteCode(" +buildingExt.getRouteCode() + "), ";
+				reportMsg += "RouteCode(" + buildingExt.getRouteCode() + "), ";
 				buildingExt.setRouteCode(archBuilding.getRouteCode());
 				buildingExt.setRouteCodeObj(routecode);
 			}
 			else {
 				buildingExt.setRouteCode("");
 			}
 		}
-		LOG.info("Updating (Name,Code,Campus,City,State,Zip,Country,Active,Address)(" + building.getBuildingName() +","+ building.getBuildingCode() +","+ building.getCampusCode() +","+ building.getBuildingAddressCityName() +","+ building.getBuildingAddressStateCode() +","+ building.getBuildingAddressZipCode() +","+ building.getBuildingAddressCountryCode() +","+ building.isActive() +","+ building.getBuildingStreetAddress() +")");
+		if (!(archibusUsaCountryCode.equalsIgnoreCase(archBuilding.getBuildingAddressCountryCode()) && kfsUsCountryCode.equalsIgnoreCase(building.getBuildingAddressCountryCode()))) {
+			reportMsg += "BuildingAddressCountryCode(" + building.getBuildingAddressCountryCode() + "), ";
+			if (archibusUsaCountryCode.equalsIgnoreCase(archBuilding.getBuildingAddressCountryCode())) {
+				building.setBuildingAddressCountryCode(kfsUsCountryCode);
+			} 
+			else {
+				building.setBuildingAddressCountryCode(archBuilding.getBuildingAddressCountryCode());
+			}
+		}
+
+		LOG.info("Updating (Name,Code,Campus,City,State,Zip,Country,Active,Address)(" + building.getBuildingName() + "," + building.getBuildingCode() + "," + building.getCampusCode() + "," + building.getBuildingAddressCityName() + "," + building.getBuildingAddressStateCode() + "," + building.getBuildingAddressZipCode() + "," + building.getBuildingAddressCountryCode() + "," + building.isActive() + "," + building.getBuildingStreetAddress() + ")");
 		buildingExt.setCampusCode(campusCode);
 		buildingExt.setBuildingCode(building.getBuildingCode());
 		businessObjectService.save(building);
 		businessObjectService.save(buildingExt);
-		writetoReportOutputFile(outReportWriter, reportMsg);
+
+		// Because Archibus country code is USA and KFS US, this does not get
+		// updated. So if a building's only difference is that Country code is
+		// US and not USA, then this does not get added to the report.
+		if (!reportMsg.matches("(?s).*Updated KFS building and overwrote these field: ")) {
+			writetoReportOutputFile(outReportWriter, reportMsg);
+		}
```

Comment:
Since no update is being generated after checking that the only difference between Archibus Building and KFS Building is Country code USA in Archibus and US in KFS, the updated info shows as blank. These lines are being removed from the report as they are not needed. Confirmation received from BA.